### PR TITLE
implement check_offset for decorations

### PR DESCRIPTION
**Pull request by nephele-gh**
This doesn't work perfeclty, there seems to be a bug still. wierdly.

thoughts?

screenshot_20230117_171633
screenshot_20230117_171651
screenshot_20230117_171758

### CODE REVIEW

File: src/mapgen/mg_decoration.cpp
Comment (SmallJoker) [inline src/mapgen/mg_decoration.cpp:123]: The bug is here: `static` arrays are initialized once in a lifetime - in this case it happens on the first decoration.
Solution: remove `static`

Also in this project, variable names should follow the snake_case convention: https://dev.minetest.net/Code_style_guidelines#Miscellaneous

File: src/script/lua_api/l_mapgen.cpp
Comment (SmallJoker) [inline src/script/lua_api/l_mapgen.cpp:1135]:
```suggestion
	if (deco->check_offset < -1 || deco->check_offset > 1) {
```

File: src/mapgen/mg_decoration.cpp
Comment (nephele-gh) [reply src/mapgen/mg_decoration.cpp:91]: I'm a bit confused about githubs UI here, it sais to commit suggestion... but it wants to create a new commit instead of updating the to-be-commited one? Seems a bit silly to commit all wrong iterations of a change?

Comment (nephele-gh) [reply src/mapgen/mg_decoration.cpp:91]: The suggested line is too long according to the guidelines

File: src/mapgen/mg_decoration.cpp
Comment (SmallJoker) [inline src/mapgen/mg_decoration.cpp:126]:
```suggestion
		const v3s16 dir_offset = v3s16(0, check_offset, 0);
```
You're adding the offset to the Y=1 values above, so to cover Y=0 with check_offset=-1, this needs correcting.

Comment (nephele-gh) [reply src/mapgen/mg_decoration.cpp]: done

File: doc/lua_api.txt
Comment (sfan5) [inline doc/lua_api.txt:9201]:
```suggestion
        check_offset = -,
        -- Specifies the offset that spawn_by should also check
        -- 0 disables additional checks, valid range: [-1, 1]
```

**Comment by sfan5** [comment]
Is there a reason we can't allow a range larger than -1 to 1? Or was this just a precaution?

File: doc/lua_api.txt
Comment (nephele-gh) [reply doc/lua_api.txt:9201]: A range to me implies a float value, hence my wording

**Comment by nephele-gh** [comment]
I've only allowed -1 and 1 mainly because this isn't implemented as a range at all, but rather just disabeling the lower check, or making it match the previous behaviour the documented described.

Is there a reason why this has to be a range? I figure one could always ammend this api later to allow more values but currently I really don't see a reason to, especially as it woulf the implementation even more complex.

**Comment by sfan5** [comment]
Oh right it's not a range at all.

File: doc/lua_api.txt
Comment (sfan5) [reply doc/lua_api.txt:9201]: well node positions are never floating point but I guess `valid values: {-1, 0, 1}` is clearer then

Comment (nephele-gh) [reply doc/lua_api.txt:9201]: I think the line mentioning -1 is the default value, and what it's use is should stay in any case, the use is not very clear and many apis /don't/ have their default values as the value set in the example of the documentation, so this would be confusing to me when not explicitly mentioned. I can reword it though if you'd like

**Review by SmallJoker — APPROVED** [review]
LGTM

**Comment by sfan5** [comment]
No, but I want to test it before approval.

**Comment by sfan5** [comment]
Please rebase. @SmallJoker feel free to merge if tested

**Comment by Zughy** [comment]
Closing because of #13152

**Pull request closed without merging**
